### PR TITLE
[API EPIC 3/4] SDK Wrapper

### PR DESCRIPTION
**Pull request by MadVikingGod**
This is a series of patches to update the metrics API.  Here is the [diff against the previous step](https://github.com/MadVikingGod/opentelemetry-go/pull/683)

1.  New API #2557 
2.  Move retained packages under SDK #2585 
3.  Create a wrapper for current SDK to be used as new API (This patch)
4.  Fix reaming tests and examples. #????

This creates a wrapper to enable the use of the current SDK as an implementation of the new API.  It does some work to remove now unneeded concepts because we access the API differently.

### CODE REVIEW

**Review by jmacd — APPROVED** [review]
:100:

**Review by Aneurysm9 — APPROVED** [review]

File: sdk/metric/sdk.go [hunk 1]
```diff
@@ -95,21 +101,15 @@ type (
...
 		// except in cases where a label set is shared due to
 		// batch recording.
-		storage attribute.Set
-
-		// labels is the processed label set for this record.
-		// this may refer to the `storage` field in another
-		// record if this label set is shared resulting from
-		// `RecordBatch`.
-		labels *attribute.Set
+		labels attribute.Set
```
Comment (dashpole) [inline sdk/metric/sdk.go:104]: nit: comment should start with "labels..."

File: sdk/metric/sdk.go [hunk 2]
```diff
@@ -239,21 +173,17 @@ func (s *syncInstrument) acquireHandle(kvs []attribute.KeyValue, labelPtr *attri
...
+	b.meter.processor.AggregatorFor(&b.descriptor, &rec.current, &rec.checkpoint)
 
 	for {
 		// Load/Store: there's a memory allocation to place `mk` into
 		// an interface here.
-		if actual, loaded := s.meter.current.LoadOrStore(mk, rec); loaded {
+		if actual, loaded := b.meter.current.LoadOrStore(mk, rec); loaded {
 			// Existing record case. Cannot change rec here because if fail
-			// will try to add rec again to avoid new allocations.
+			// will try to add rec again to avoid new allocationb.
```
Comment (dashpole) [inline sdk/metric/sdk.go:186]: s/allocationb/allocations

File: sdk/metric/sdk.go [hunk 3]
```diff
@@ -280,9 +210,16 @@ func (s *syncInstrument) acquireHandle(kvs []attribute.KeyValue, labelPtr *attri
...
 // The order of the input array `kvs` may be sorted after the function is called.
 func (s *syncInstrument) RecordOne(ctx context.Context, num number.Number, kvs []attribute.KeyValue) {
-	h := s.acquireHandle(kvs, nil)
+	h := s.acquireHandle(kvs)
 	defer h.unbind()
-	h.RecordOne(ctx, num)
+	h.captureOne(ctx, num)
+}
+
+// @@@
```
Comment (dashpole) [inline sdk/metric/sdk.go:218]: TODO?

**Review by dashpole — APPROVED** [review]
Just nits.  Probably should be handled later.

**Comment by MadVikingGod** [comment]
I was going to pull these changes into one on the end if you are ok with that @dashpole.

File: sdk/metric/sdk.go [hunk 4]
```diff
@@ -296,35 +233,58 @@ func (s *syncInstrument) RecordOne(ctx context.Context, num number.Number, kvs [
...
 			meter:      m,
 		},
 	}
-	m.asyncLock.Lock()
-	defer m.asyncLock.Unlock()
-	m.asyncInstruments.Register(a, runner)
 	return a, nil
 }
 
+func (m *Accumulator) RegisterCallback(insts []instrument.Asynchronous, f func(context.Context)) error {
```
Comment (MrAlias) [inline sdk/metric/sdk.go:264]: Missing doc string.

**Review by MrAlias — APPROVED** [review]

**Pull request closed without merging**